kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java

```diff
@@ -340,7 +341,13 @@ public void setLimitSerializationClasses(Boolean limitSerializationClasses) {
         return input;
     }
 
-    protected void removeTransient(Set<?> input) {
+    protected List<ObjectModel> cleanList(List<ObjectModel> input) {
+        input.remove(null);
```

Comment:
For removing all nulls (if more than one), it should be a while loop:
```suggestion
        while (input.remove(null));
```

Reply:
done